src/category_theory/monoidal/Mon_.lean

```diff
@@ -273,6 +297,158 @@ def equiv_lax_monoidal_functor_punit : lax_monoidal_functor (discrete punit.{u+1
+  slice_lhs 1 2 { rw [←left_unitor_tensor_inv] },
+  rw ←(cancel_epi (λ_ (𝟙_ C)).inv),
+  slice_lhs 1 2 { rw [left_unitor_inv_naturality] },
+  simp only [category.assoc],
+end
+
+lemma mul_associator {M N P : Mon_ C} :
+    (tensor_μ C (M.X ⊗ N.X, P.X) (M.X ⊗ N.X, P.X) ≫
+      (tensor_μ C (M.X, N.X) (M.X, N.X) ≫ (M.mul ⊗ N.mul) ⊗ P.mul)) ≫
+    (α_ M.X N.X P.X).hom
+  = ((α_ M.X N.X P.X).hom ⊗ (α_ M.X N.X P.X).hom) ≫
+    tensor_μ C (M.X, N.X ⊗ P.X) (M.X, N.X ⊗ P.X) ≫
+    (M.mul ⊗ tensor_μ C (N.X, P.X) (N.X, P.X) ≫ (N.mul ⊗ P.mul)) :=
+begin
+  simp,
+  slice_lhs 2 3 { rw [←category.id_comp P.mul, tensor_comp] },
+  slice_lhs 3 4 { rw [associator_naturality] },
+  slice_rhs 3 4 { rw [←category.id_comp M.mul, tensor_comp] },
+  slice_lhs 1 3 { rw associator_monoidal },
+  simp only [category.assoc],
+end
+
+lemma one_left_unitor {M : Mon_ C} :
+  ((λ_ (𝟙_ C)).inv ≫ (𝟙 (𝟙_ C) ⊗ M.one)) ≫ (λ_ M.X).hom = M.one :=
+by { slice_lhs 2 3 { rw left_unitor_naturality }, simp }
+
+lemma mul_left_unitor {M : Mon_ C}:
+    (tensor_μ C (𝟙_ C, M.X) (𝟙_ C, M.X) ≫ ((λ_ (𝟙_ C)).hom ⊗ M.mul)) ≫ (λ_ M.X).hom
+  = ((λ_ M.X).hom ⊗ (λ_ M.X).hom) ≫ M.mul :=
+begin
+  rw [←(category.comp_id (λ_ (𝟙_ C)).hom), ←(category.id_comp M.mul), tensor_comp],
+  slice_lhs 3 4 { rw left_unitor_naturality },
+  slice_lhs 1 3 { rw ←left_unitor_monoidal },
+  simp only [category.assoc, category.id_comp],
+end
+
+lemma one_right_unitor {M : Mon_ C} :
+  ((λ_ (𝟙_ C)).inv ≫ (M.one ⊗ 𝟙 (𝟙_ C))) ≫ (ρ_ M.X).hom = M.one :=
+by { slice_lhs 2 3 { rw [right_unitor_naturality, ←unitors_equal] }, simp }
+
+lemma mul_right_unitor {M : Mon_ C} :
+    (tensor_μ C (M.X, 𝟙_ C) (M.X, 𝟙_ C) ≫ (M.mul ⊗ (λ_ (𝟙_ C)).hom)) ≫ (ρ_ M.X).hom
+  = ((ρ_ M.X).hom ⊗ (ρ_ M.X).hom) ≫ M.mul :=
+begin
+  rw [←(category.id_comp M.mul), ←(category.comp_id (λ_ (𝟙_ C)).hom), tensor_comp],
+  slice_lhs 3 4 { rw right_unitor_naturality },
+  slice_lhs 1 3 { rw ←right_unitor_monoidal },
+  simp only [category.assoc, category.id_comp],
+end
+
+instance Mon_monoidal : monoidal_category (Mon_ C) :=
+{ tensor_obj := λ M N,
+  { X := M.X ⊗ N.X,
+    one := (λ_ (𝟙_ C)).inv ≫ (M.one ⊗ N.one),
+    mul := tensor_μ C (M.X, N.X) (M.X, N.X) ≫ (M.mul ⊗ N.mul),
+    one_mul' := Mon_tensor_one_mul C M N,
+    mul_one' := Mon_tensor_mul_one C M N,
+    mul_assoc' := Mon_tensor_mul_assoc C M N },
+  tensor_hom := λ M N P Q f g,
+  { hom := f.hom ⊗ g.hom,
+    one_hom' :=
+    begin
+      dsimp,
+      slice_lhs 2 3 { rw [←tensor_comp, hom.one_hom f, hom.one_hom g] },
+    end,
+    mul_hom' :=
+    begin
+      dsimp,
+      slice_rhs 1 2 { rw [tensor_μ_natural] },
+      slice_lhs 2 3 { rw [←tensor_comp, hom.mul_hom f, hom.mul_hom g, tensor_comp] },
+      simp only [category.assoc],
+    end },
+  tensor_id' := by { intros, ext, apply tensor_id },
+  tensor_comp' := by { intros, ext, apply tensor_comp },
+  tensor_unit := trivial C,
+  associator := λ M N P,
+  iso_of_iso
+    (α_ M.X N.X P.X)
+    (by { dsimp, apply one_associator })
```

Comment:
`dsimp` is not necessary here or below, so in fact you can avoid even having `by` block: just stay in term mode. Even better, change the argument of `one_associator` to `{C}`.

Similarly below.